nixos/modules/services/security/endlessh-go.nix

```
@@ -0,0 +1,96 @@
+{ config
+, lib
+, pkgs
+, ...
+}:
+
+with lib;
+
+let
+  cfg = config.services.endlessh-go;
+in
+{
+  options.services.endless-go = {
+    enable = mkEnableOption "Whether to enable the endlessh-go service.";
+
+    listenAddress = mkOption {
+      type = types.str;
+      default = "0.0.0.0";
+      description = mdDoc ''
+        Interface address to bind the endlessh-go daemon to SSH connections.
+      '';
+    };
+
+    port = mkOption {
+      type = types.port;
+      default = 2222;
+      description = mdDoc ''
+        Specifies on which port the endlessh-go daemon listens for SSH
+        connections.
+      '';
+    };
+
+    prometheus = {
+      enable = mkEnableOption "Whether to enable Prometheus integration.";
+
+      listenAddress = mkOption {
+        type = types.str;
+        default = "0.0.0.0";
+        description = mdDoc ''
+          Interface address to bind the endlessh-go daemon to answer Prometheus
+          queries.
+        '';
+      };
+
+      port = mkOption {
+        type = types.port;
+        default = 2112;
+        description = mdDoc ''
+          Specifies on which port the endlessh-go daemon answers for Prometheus
+          queries.
+        '';
+      };
+    };
+
+    extraOptions = mkOption {
+      type = with types; nullOr (listOf str);
```

```suggestion
      type = with types; listOf str;
```
empty lists already have "no such thing" semantics